array_test.go

```diff
@@ -604,78 +604,245 @@ func TestArray_IsEqualUnordered(t *testing.T) {
 		value.chain.assertFailed(t)
 		value.chain.clearFailed()
 	})
+
+	t.Run("canonization", func(t *testing.T) {
+		type (
+			myArray []interface{}
+			myInt   int
+		)
+
+		value := NewArray(reporter, []interface{}{123, 456, "foo"})
+		duplicateValue := NewArray(reporter, []interface{}{123, 123, "foo"})
+
+		assert.Equal(t, []interface{}{123.0, 456.0, "foo"}, value.Raw())
+
+		value.IsEqualUnordered(myArray{myInt(456), 123.0, "foo"})
+		value.chain.assertNotFailed(t)
+		value.chain.clearFailed()
+
+		value.NotEqualUnordered(myArray{myInt(456), 123.0, "foo"})
+		value.chain.assertFailed(t)
+		value.chain.clearFailed()
+
+		value.IsEqualUnordered(myArray{456.0, myInt(123.0), "foo"})
+		value.chain.assertNotFailed(t)
+		value.chain.clearFailed()
+
+		value.NotEqualUnordered(myArray{456.0, myInt(123.0), "foo"})
+		value.chain.assertFailed(t)
+		value.chain.clearFailed()
+
+		value.IsEqualUnordered(myArray{"123", "456", "foo"})
+		value.chain.assertFailed(t)
+		value.chain.clearFailed()
+
+		value.NotEqualUnordered(myArray{"123", "456", "foo"})
+		value.chain.assertNotFailed(t)
+		value.chain.clearFailed()
+
+		value.IsEqualUnordered(myArray{"123.0", "456.0", "foo"})
+		value.chain.assertFailed(t)
+		value.chain.clearFailed()
+
+		value.NotEqualUnordered(myArray{"123.0", "456.0", "foo"})
+		value.chain.assertNotFailed(t)
+		value.chain.clearFailed()
+
+		value.IsEqualUnordered(nil)
+		value.chain.assertFailed(t)
+		value.chain.clearFailed()
+
+		value.NotEqualUnordered(nil)
+		value.chain.assertFailed(t)
+		value.chain.clearFailed()
```

Comment:
This is a good check, but I suggest to extract it into a separate subtest, like, "nil argument" or "invalid argument", because it's more like a special case.

I understand that you likely modeled it after TestArray_IsEqual/canonization, but actually we need to do the same there too (you can do it or I'll do it later, as you wish).

Reply:
Same for `array.InList(func() {})` and similar tests

Reply:
I've segmented the nil and invalid arguments into their own tests